[KIECLOUD-593-7.12.x] Update BA Operator Docs about OLM (#696)
kiegroup/kie-cloud-operator: README.md

docker push quay.io/kiegroup/kie-cloud-operator:<version>
```

## Deploy to OpenShift 4.7+ using OLM

To install this operator on OpenShift 4 for end-to-end testing, make sure you have access to a quay.io (https://quay.io/) account to create
an application repository. Follow the [authentication](https://github.com/operator-framework/operator-courier/#authentication)
instructions for Operator Courier to obtain an account token.
This token is in the form of "basic XXXXXXXXX" and both words are required for the command.

Also note that the push command does not overwrite an existing repository,
and it needs to be deleted before a new version can be built and uploaded.
Once the bundle has been uploaded, create an [Operator Source](https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md#linking-the-quay-application-repository-to-your-openshift-40-cluster)
to load your operator bundle in OpenShift.

> Requires [opm](https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/latest/)

Reviewer: Here you are replacing link to github release of opm command with link to ocp mirror of ocp client (latest version). I'm not sure if opm is part of ocp cli. I think it would be better to keep this link as it was.

Author: ok

Requires [podman](https://podman.io/)

Requires [cekit](https://github.com/cekit/cekit)

> **Create your own Bundle**
> Remove the following line from deploy/olm-catalog/dev/<last-version>/manifest/businessautomation-operator.clusterserviceversion.yaml

Reviewer: there is needed empty line, otherwise title and text will be on the same line
```suggestion
**Create your own Bundle**

Remove the following line from deploy/olm-catalog/dev/<last-version>/manifest/businessautomation-operator.clusterserviceversion.yaml
```

> Remove the following line from deploy/olm-catalog/dev/<last-version>/manifest/businessautomation-operator.clusterserviceversion.yaml

Reviewer: part <last-version> is not replaced in doc, it would be better to use soma version as example.
Also it would be good to mention, that if this operator is not available on local or is not removed, then build of indexing image will fail. And that this step is only for dev builds.

Author: ok

```console
replaces: businessautomation-operator.<last-version>
```

Activate Cekit and run the following command
```console
$ make bundle-dev
```

the last log line is something like this:
```console
INFO  Image built and available under following tags: quay.io/<your_quay_username>/rhpam-operator-bundle:7.12.1, quay.io/${USERNAME}/rhpam-operator-bundle:latest
```
**Push the bundle on the container registry**

VERSION=$(go run getversion.go)

USERNAME=<your_quay_username>
```console
$ docker push quay.io/${USERNAME}/rhpam-operator-bundle:${VERSION}
```

#### Build the index
```console

> opm index add --bundles quay.io/${USERNAME}/rhpam-operator-bundle:${VERSION} --tag quay.io/${USERNAME}/rhpam-operator--index:${VERSION} --permissive

Reviewer: fix naming and remove permissive parameter
```suggestion
opm index add --bundles quay.io/${USERNAME}/rhpam-operator-bundle:${VERSION} --tag quay.io/${USERNAME}/rhpam-operator-index:${VERSION}
```

```
#### Push the index on the container registry
Podman images
```console

> podman push quay.io/${USERNAME}/rhpam-operator--index:${VERSION}

Reviewer: ```suggestion
podman push quay.io/${USERNAME}/rhpam-operator-index:${VERSION}
```

Author: ok

```

#### Disable default catalog sources on Openshift
To test your Operator, with bundle and index you need to disable the default source like the operator hub
```console
oc patch OperatorHub cluster --type json -p '[{"op": "add", "path": "/spec/disableAllDefaultSources", "value": true}]'
```

#### catalog-source.yaml
```console
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: xxxxxname
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: xxxxximage
  displayName: My Operator Catalog
  publisher: grpc
```

> Choose a CATALOG_SOURCE_NAME something like "max-operator-manifests"

Reviewer: ```suggestion
Choose a CATALOG_SOURCE_NAME something like "my-operator-manifests"
```

```console
sed -i "s/xxxxxname/$CATALOG_SOURCE_NAME/" catalog-source.yaml
```
```console
sed -i "s/xxxxximage/$(echo $INDEXING_IMAGE | sed "s.\/.\\\/.g")/" catalog-source.yaml

> ```

Reviewer: I wouldn't keep there these sed commands.
Instead, it would be better to have there the catalog-source.yaml with already replaced values. There can be additional comments to this file, what is needed to be replaced there - the image reference.

Example of catalog-source
```console
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: max-operator-manifests
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/<your_quay_id>/rhpam-operator--index:7.12.1
  displayName: My Operator Catalog
  publisher: grpc
```

#### Create catalog source on Openshift
```console
oc create -f catalog-source.yaml
```

> #### subscription.yaml

Reviewer: It may not be clear for what is subscription needed, please add more comments.
There can be also mentioned, that you can use Operator Hub to list your operators and create subscriptions.

```console
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: businessautomation-operator
  namespace: <your-namespace>
spec:
  channel: stable
  name: businessautomation-operator
  source: $CATALOG_SOURCE_NAME
  sourceNamespace: openshift-marketplace
```

Example of subscription

```console
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: businessautomation-operator
  namespace: max
spec:
  channel: stable
  name: businessautomation-operator
  source: max-operator-manifests
  sourceNamespace: openshift-marketplace
```

#### Create subscription
```console
oc create -f subscription.yaml
```

> On Openshift go to project "openshift-marketplace" to see your subscription and your operator,

Reviewer: ```suggestion
On OpenShift go to your project (e.g. max) to see your subscription and your operator,
```

Author: ok
